rusk/src/lib/lib.rs

```diff
@@ -50,6 +52,20 @@ pub static PUB_PARAMS: Lazy<PublicParameters> = Lazy::new(|| unsafe {
 
 const STREAM_BUF_SIZE: usize = 64;
 
+/// The gas grace is a magic number denoting the amount of gas that is spent by
+/// a transaction after the charging code has been executed.
+const GAS_GRACE: u64 = 24_858_000;
+
+/// Computes the gas limit to apply to a transaction, given the limit imposed by
+/// the fee limit and the block gas limit.
+///
+/// The result will be the minimum between the block gas limit and the fee limit
+/// plus the [`GAS_GRACE`].
+fn compute_gas_limit(fee_limit: u64, block_limit: u64) -> u64 {
+    let min = cmp::min(fee_limit, block_limit);
+    min + GAS_GRACE
+}
```

Comment:
Adding the GAS_GRACE to the gas_limit used by a TX, it's not a good idea IMO.

IIUC the need for this additional GAS, is to have GAS to execute the charging operation.
Adding it to the whole GasLimit may result in having this "special gas amount" fully spent by a TX that is going OOG.

I strongly believe we should find another way to add this "reserved" amount of gas:
- Should we explicitly add a `reserved_gas_amount` that is not passed to the inner calls?
- Is there a way to flag the charging code to no require GAS at all?

Reply:
Answering the first bullet: yes, it is possible. In fact, we have this feature releasing out of `piecrust` today. As for the second, I'm not sure. It could be possible by modifying the metering middleware, but we should be careful that this feature is not available to anyone else.

Reply:
Should we wait to apply `piecrust@0.6` to apply this feature?

Reply:
It's ok, we can add it even right now if there is a tracking issue to be tackled later on